backend/doc/migration.md

```diff
@@ -34,7 +34,397 @@ and activate the Tracim virtualenv:
+    # for ck contraint.
+    # "ck": "ck_%(table_name)s_%(constraint_name)s",
+    "fk": "fk_%(table_name)s_%(column_0_name)s_%(referred_table_name)s",
+    "pk": "pk_%(table_name)s",
+}
+
+metadata = MetaData(naming_convention=NAMING_CONVENTION)
+DeclarativeBase = declarative_base(metadata=metadata)
+
+
+class TemporaryContentRevision(DeclarativeBase):
+    """ temporary sqlalchemy object to help migration"""
+
+    __tablename__ = "content_revisions"
+
+    revision_id = Column(Integer, primary_key=True)
+    properties = Column("properties", Text(), unique=False, nullable=False, default="")
+    type = Column(Unicode(32), unique=False, nullable=False)
+
+
+def upgrade():
+    connection = op.get_bind()
+    session = Session(bind=connection)
+    # check all revision with not empty content
+    revisions = session.query(TemporaryContentRevision).filter(
+        TemporaryContentRevision.properties != ""
+    )
+    # for each revision go to json properties['allowed_content'] and search for
+    # OLD_SLUG section
+    for rev in revisions:
+        if rev.properties:
+            json_properties = json.loads(rev.properties)
+            allowed_content_properties = json_properties.get("allowed_content")
+            if allowed_content_properties:
+                if OLD_SLUG in allowed_content_properties:
+                    if NEW_SLUG not in allowed_content_properties:
+                        # add old value to new slug
+                        allowed_content_properties[NEW_SLUG] = allowed_content_properties[OLD_SLUG]
+                    # remove old slug section
+                    del allowed_content_properties[OLD_SLUG]
+                    # convert to json and apply modification
+                    new_properties = json.dumps(json_properties)
+                    rev.properties = new_properties
+                    session.add(rev)
+    session.commit()
+```
+
+see :
+- [354d62d490ad_remove_page_slug_from_properties.py](../tracim_backend/migration/354d62d490ad_remove_page_slug_from_properties.py)
+- [cd79614189ac_add_owner_to_workspace.py](../tracim_backend/migration/cd79614189ac_add_owner_to_workspace.py)
+
+
+#### Use Raw SQL
+
+In some specific case, using raw sql query is the easiest solution, be careful about
+using sql that does work on all your supported database.
+
+
+```python
+from alembic import op
+
+filling_current_revision_query = """
+update content
+set revision_id = (select max(revision_id) from content_revisions where content_id = content.id)
+"""
+
+def upgrade():
+    connection = op.get_bind()
+    connection.execute(filling_current_revision_query)
+
+```
+
+### Doing specific database code
+
+to make specific code for some database, we can rely on sqlalchemy dialect:
+
+```python
+from alembic import op
+dialect = op.get_context().dialect
+if dialect.name == "mysql":
+   pass
+```
+
+in mysql case, dialect does not tell all the true, you can use specific method to
```

Comment:
In the case of MySQL

What do you mean by "all the true"?

Reply:
Mysql Dialect, is same for both mysql and mariadb and those 2 software are not compatible anymore, especially since mysql 8.